src/TinyHelpers/Extensions/CollectionExtensions.cs

```diff
@@ -227,4 +227,25 @@ public static IEnumerable<TSource> WhereIf<TSource>(this IEnumerable<TSource> so
     /// <returns>An <see cref="IQueryable{T}"/> that contains elements from the input sequence that satisfy the condition specified by <paramref name="predicate"/>; the unfiltered <paramref name="source"/>, otherwise.</returns>
     public static IQueryable<TSource> WhereIf<TSource>(this IQueryable<TSource> source, bool condition, Expression<Func<TSource, bool>> predicate)
         => condition ? source.Where(predicate) : source;
+
+    /// <summary>
+    /// Splits the source collection into chunks of a specified size.
+    /// </summary>
+    /// <typeparam name="TSource">The type of the elements in the source collection.</typeparam>
+    /// <param name="source">The source collection to split into chunks.</param>
+    /// <param name="chunkSize">The size of each chunk.</param>
+    /// <returns>An <see cref="IEnumerable{T}"/> where each element is an array of elements of the specified chunk size.</returns>
+    /// <exception cref="ArgumentOutOfRangeException"><paramref name="chunkSize"/> is less than 1.</exception>
+    public static IEnumerable<IEnumerable<TSource>> ChunkBySize<TSource>(this IEnumerable<TSource> source, int chunkSize)
```

Comment:
Starting from version 6.0, .NET already provides a `Chunk` extension method for `IEnumerable`: https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.chunk.

You could rename this method as `Chunk` and specify a conditional compilation condition to add this method only for .NET Standard 2.0.

Reply:
Sounds good, `Chunk` method bounded to standard .NET Standard 2.0